fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/InclusionQuerySegmentAggregator.java

```diff
@@ -199,23 +199,41 @@ private void processIncludeParameters(StringBuilder queryString, List<Object> bi
             // R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID AND
             queryString.append("R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID AND ");
             // ('Organization/' || LR.LOGICAL_ID IN 
-            queryString.append("('").append(includeParm.getSearchParameterTargetType()).append("/' || LR.LOGICAL_ID IN ");
-            // (SELECT P1.STR_VALUE FROM OBSERVATION_STR_VALUES P1 WHERE
-            queryString.append("(SELECT P1.STR_VALUE FROM ").append(this.resourceType.getSimpleName()).append("_STR_VALUES P1 WHERE ");
+            queryString.append("('").append(includeParm.getSearchParameterTargetType()).append("/' || LR.LOGICAL_ID IN ("); 
+
+            // Run the following sub Query only once to improve performance.
+            StringBuilder subQueryString = new StringBuilder();
+            // SELECT P1.STR_VALUE FROM OBSERVATION_STR_VALUES P1 WHERE
+            subQueryString.append("SELECT P1.STR_VALUE FROM ").append(this.resourceType.getSimpleName()).append("_STR_VALUES P1 WHERE ");
             // P1.PARAMETER_NAME_ID=xx AND 
-            queryString.append("P1.PARAMETER_NAME_ID=").append(this.getParameterNameId(includeParm.getSearchParameter())).append(" AND ");
+            subQueryString.append("P1.PARAMETER_NAME_ID=").append(this.getParameterNameId(includeParm.getSearchParameter())).append(" AND ");
             // P1.RESOURCE_ID IN 
-            queryString.append("P1.LOGICAL_RESOURCE_ID IN ");
+            subQueryString.append("P1.LOGICAL_RESOURCE_ID IN ");
             // (SELECT R.LOGICAL_RESOURCE_ID  
-            queryString.append("(SELECT R.LOGICAL_RESOURCE_ID ");
+            subQueryString.append("(SELECT R.LOGICAL_RESOURCE_ID ");
             // Add FROM clause for "root" resource type
-            queryString.append(super.buildFromClause());
+            subQueryString.append(super.buildFromClause());
             // Add WHERE clause for "root" resource type
-            queryString.append(super.buildWhereClause());
-            
-            queryString.append(")))");
-                        
-            this.addBindVariables(bindVariables);
+            subQueryString.append(super.buildWhereClause());
+            subQueryString.append(")");    
+
+            // ('Patient/326faf57-2aff-4ae5-ab41-87eace6c8f33')
+            SqlQueryData subQueryData = new SqlQueryData(subQueryString.toString(), bindVariables);
+            boolean isFirstItem = true;
+            for (String strValue: this.resourceDao.searchStringValues(subQueryData)) {
+                if (!isFirstItem) {
+                    queryString.append(" , ");
+                }
+                if (strValue != null) {
+                    queryString.append("'").append(strValue).append("'");
+                    isFirstItem = false;
+                }
+            }
```

Comment:
I suggest splitting this whole section into a new method with a name like `executeIncludeSubQuery`